build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1185,6 +1417,156 @@ message ExecuteRequest {
+  Digest action_digest = 2;
+
+  // A series of incremental (streamed) stdout and stderr outputs in order of
+  // write from the action during execution. The full set of console outputs
+  // from the action will also be written in the final
+  // [ExecuteResponse][build.bazel.remote.execution.v2.ExecuteResponse].
+  // The server MAY buffer these outputs in order to return larger batches
+  // of messages less frequently. The server MAY choose to omit incremental
+  // messages and instead wait to the final `ExecuteResponse` to return all
+  // outputs from the action, however in this case the server MUST NOT return
+  // any incremental messages in this field.
+  ConsoleOutputs streamed_console_outputs = 3;
+
+  // A directory path for which the worker does not have metadata. This
+  // indicates a directory that the `Action` was not predicted to use
+  // from the client filesystem. The client is requested to enumerate
+  // directory metadata and respond in
+  // [StreamedDynamicExecutionData.dynamic_directory_metadata_response][build.bazel.remote.execution.StreamedDynamicExecutionData].
+  // The client MAY include individual file digests but this is not required,
+  // the worker will request individual file hashes as needed using
+  // `dynamic_file_hash_path`.
+  // Depending on the server and client implementation this may be a
+  // relative or fully qualified path.
+  string dynamic_directory_content_path = 4;
+
+  // A file path for which the worker does not have a hash. This indicates
+  // a file that the `Action` was not predicted to use from the client
+  // filesystem. The client is requested to hash the file and respond in
+  // [StreamedDynamicExecutionData.dynamic_file_hash_response][build.bazel.remote.execution.StreamedDynamicExecutionData].
+  // Depending on the server and client implementation this may be a
+  // relative or fully qualified path.
+  string dynamic_file_hash_path = 5;
+
+  // A file path and expected content hash that the worker does not have in
+  // the CAS. This indicates file content that the `Action` was not predicted
+  // to use from the client, or content that was subsequently removed from
+  // the CAS. The client is requested to upload the file content into the
+  // CAS and respond in 
+  // [StreamedDynamicExecutionData.dynamic_file_content_response][build.bazel.remote.execution.StreamedDynamicExecutionData].
+  DynamicFileContentRequest dynamic_file_content_request = 6;
+}
+
+// A streaming request message for
+// [Execution.ExecuteWithDynamicRetrieval][build.bazel.remote.execution.v2.Execution.ExecuteWithDynamicRetrieval].
+message StreamedDynamicExecutionData {
+  // Sent in response to a dynamic directory metadata request.
+  message DynamicDirectoryMetadataResponse {
+    // The directory path, copied from the directory metadata request.
+    string path = 1;
+
+    // Metadata for the directory. [DirectoryNode][build.bazel.remote.execution.v2.Execution.DirectoryNode]
```

Comment:
I think there's enough use of metadata in this API that we should avoid introducing another one.

Reply:
Quite. Changed to directory content from directory metadata in message and field names.